### PR TITLE
[doc] Add V2/V2S checklists and declare V2S

### PR DESCRIPTION
**Pull request by GregAC**
There's still some actions outstanding from V2/V2S but it's getting close.

I wanted to get this out as PR so there's a chance for people to raise objections to the checklist or the way I've presented this within the Ibex documentation.

We still need some OT doc updates to point to the Ibex documentation for the V2/V2S checklists.

Merging this PR will declare V2/V2S for Ibex.

### CODE REVIEW

**Comment by GregAC** [comment]
You can see a rendered version here: https://ibex-core--1939.org.readthedocs.build/en/1939/03_reference/verification.html#verification-stages

File: doc/03_reference/verification.rst [hunk 1]
```diff
@@ -223,6 +223,138 @@ You can add any compile/runtime options in `dv/uvm/core_ibex/yaml/simulator.yaml
...
+|               |                                     |            | Publicly viewable reports on the `OpenTitan regression dashboard <https://reports.opentitan.org/hw/top_earlgrey/dv/summary/latest/report.html>`_ are planned for V3. |
++---------------+-------------------------------------+------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Coverage      | SIM_CODE_COVERAGE_V2                | Complete   | Coverage results available in nightly regression run.                                                                                                                |
++---------------+-------------------------------------+------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Coverage      | SIM_FUNCTIONAL_COVERAGE_V2          | Complete   | Coverage results avaialble in nightly regression run.                                                                                                                |
```
Comment (andreaskurth) [inline doc/03_reference/verification.rst:264]:
```suggestion
| Coverage      | SIM_FUNCTIONAL_COVERAGE_V2          | Complete   | Coverage results available in nightly regression run.                                                                                                                |
```

File: doc/03_reference/verification.rst [hunk 2]
```diff
@@ -223,6 +223,138 @@ You can add any compile/runtime options in `dv/uvm/core_ibex/yaml/simulator.yaml
...
+|               |                                     |            | Note the average grade (the average of coverage % for each individual coverpoint and cross) is used for the 90% figure.                                              |
+|               |                                     |            | As the functional coverage contains some very large crosses as simple % of all bins hit biases too much toward these crosses.                                        |
```
Comment (andreaskurth) [inline doc/03_reference/verification.rst:266]:
```suggestion
|               |                                     |            | As the functional coverage contains some very large crosses a simple % of all bins hit biases too much toward these crosses.                                        |
```
not sure?

File: doc/03_reference/verification.rst [hunk 3]
```diff
@@ -223,6 +223,138 @@ You can add any compile/runtime options in `dv/uvm/core_ibex/yaml/simulator.yaml
...
+""""""""""""""""""
+
++---------------+--------------------------+------------+--------------------------------------------------------------------------------------------------------------------------------------+
+| Type          | Item                     | Resolution | Notes                                                                                                                                |
++===============+==========================+============+======================================================================================================================================+
+| Documentation | SEC_CM_TESTPLAN_COMPLETE | Complete   | Security counter measure to test mapping can be found below                                                                          |
++---------------+--------------------------+------------+--------------------------------------------------------------------------------------------------------------------------------------+
+| Tests         | FPV_SEC_CM_VERIFIED      | Complete   | See the `OpenTitan FPV Results Dashboard <https://reports.opentitan.org/hw/top_earlgrey/formal/sec_cm/summary/latest/report.html>`_. |
```
Comment (andreaskurth) [inline doc/03_reference/verification.rst:299]: (Should soon show up on dashboard, now that https://github.com/lowRISC/opentitan/pull/16156 is merged.)

File: doc/03_reference/verification.rst [hunk 4]
```diff
@@ -223,6 +223,138 @@ You can add any compile/runtime options in `dv/uvm/core_ibex/yaml/simulator.yaml
...
+|                                | The ``chip_sw_data_integrity`` OpenTitan top-level test will trigger integrity errors within the OpenTitan specific ``rv_core_ibex`` wrapper.                        |
+|                                | The TL-UL host adapter used in the OpenTitan specific ``rv_core_ibex`` is fully verified elsewhere in OpenTitan.                                                     |
++--------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| SCRAMBLE.KEY.SIDELOAD          | ``riscv_rand_instr_test`` in Ibex DV.                                                                                                                                |
+|                                | This test executes ``FENCE.I`` which rotates the scramble key                                                                                                        |
```
Comment (andreaskurth) [inline doc/03_reference/verification.rst:323]: Suggest to mention the recently added `rv_core_ibex_icache_invalidate_test` OT TLT (consistent with `BUS.INTEGRITY`) along the following lines:
> The `rv_core_ibex_icache_invalidate_test` OpenTitan top-level test covers assertions within the OpenTitan specific `rv_core_ibex` wrapper that check that a `FENCE.I` results in an icache scramble key request and that the returned key is correctly supplied to the scrambling memory primitives.

File: doc/03_reference/verification.rst [hunk 5]
```diff
@@ -223,6 +223,138 @@ You can add any compile/runtime options in `dv/uvm/core_ibex/yaml/simulator.yaml
...
+|                                | This test produces double faults, which are checked by an assertion.                                                                                                 |
+|                                | ``chip_sw_rv_core_ibex_double_fault`` top-level test in OpenTitan demonstrates escalation on a double fault                                                          |
++--------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| ICACHE.MEM.SCRAMBLE            | No explicit testing, the scrambling memory primitive is seperately verified within OpenTitan.                                                                        |
+|                                | Assertions in the OpenTitan specific ``rv_core_ibex`` wrapper ensure a newly requested scramble key is correctly applied to the scrambling memories.                 |
```
Comment (andreaskurth) [inline doc/03_reference/verification.rst:354]: Suggest to reference `SCRAMBLE.KEY.SIDELOAD` for details

**Review by andreaskurth — COMMENTED** [review]
Thanks a lot for bringing it all together, @GregAC!

File: doc/03_reference/verification.rst [hunk 6]
```diff
@@ -223,6 +223,138 @@ You can add any compile/runtime options in `dv/uvm/core_ibex/yaml/simulator.yaml
...
+| Review        | V3_CHECKLIST_SCOPED                 | Complete   |                                                                                                                                                                      |
++---------------+-------------------------------------+------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+
+**PMP Testing Note**: A large number of iterations of ``pmp_full_random_test`` are required to meet coverage goals and timed out tests must be included in the coverage collection.
+This is because of the large cross bins for PMP that aim to explore the full space of possible behaviour.
+The current strategy of random generation is very inefficient at exploring this space.
+It is also complex to write a randomly generated test that can deal with all possible scenarios without hitting a double faulting or time out scenarios (e.g. consider a random configuration that gives you no executable regions).
```
Comment (marnovandermaas) [inline doc/03_reference/verification.rst:286]: This is especially true when enhanced PMP modes like machine mode lockdown and machine mode whitelist policy are enabled.

File: doc/03_reference/verification.rst [hunk 7]
```diff
@@ -223,6 +223,138 @@ You can add any compile/runtime options in `dv/uvm/core_ibex/yaml/simulator.yaml
...
+For future work we will explore more efficient strategies for exploring this space as well as employing formal methods to achieve full verification closure.
+
+Ibex V2S Checklist
+""""""""""""""""""
+
++---------------+--------------------------+------------+--------------------------------------------------------------------------------------------------------------------------------------+
+| Type          | Item                     | Resolution | Notes                                                                                                                                |
++===============+==========================+============+======================================================================================================================================+
+| Documentation | SEC_CM_TESTPLAN_COMPLETE | Complete   | Security counter measure to test mapping can be found below                                                                          |
```
Comment (marnovandermaas) [inline doc/03_reference/verification.rst:297]:
```suggestion
| Documentation | SEC_CM_TESTPLAN_COMPLETE | Complete   | The security counter measure to test mapping can be found below                                                                          |
```

**Review by marnovandermaas — APPROVED** [review]
LGTM, just some minor comments.

**Comment by GregAC** [comment]
Thanks for the feedback @marnovandermaas and @andreaskurth I have incorporated your comments.

File: doc/03_reference/verification.rst [hunk 8]
```diff
@@ -223,6 +223,139 @@ You can add any compile/runtime options in `dv/uvm/core_ibex/yaml/simulator.yaml
...
++---------------+-------------------------------------+------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Testbench     | SIM_TB_ENV_COMPLETED                | Complete   |                                                                                                                                                                      |
++---------------+-------------------------------------+------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Tests         | SIM_ALL_TESTS_PASSING               | Complete   | Note the ``riscv_assorted_traps_interrupts_debug`` test sees many failures (but does have some seeds that pass).                                                     |
+|               |                                     |            | The test attempts to generally combine many different stimuluses and under OpenTitan classification would be considered a V3 test.                                   |
```
Comment (msfschaffner) [inline doc/03_reference/verification.rst:251]: stimuluses -> stimuli :)?

File: doc/03_reference/verification.rst [hunk 9]
```diff
@@ -223,6 +223,139 @@ You can add any compile/runtime options in `dv/uvm/core_ibex/yaml/simulator.yaml
...
+This is because of the large cross bins for PMP that aim to explore the full space of possible behaviour.
+The current strategy of random generation is very inefficient at exploring this space.
+It is also complex to write a randomly generated test that can deal with all possible scenarios without hitting a double faulting or time out scenarios (e.g. consider a random configuration that gives you no executable regions and ePMP modes like machine mode lockdown and machine mode whitelist policy).
+Co-simulation checking is enabled when this test is run (as it is for all block level verification tests) so would detect any incorrect behaviour.
+From investigation we are confident the time-outs seen are simply badly performing tests (e.g. very slowly working its way through an instruction block with no execute permissions by attempting to execute one instruction, faulting, trying the next and getting the same result over and over).
+For future work we will explore more efficient strategies for exploring this space as well as employing formal methods to achieve full verification closure.
```
Comment (msfschaffner) [inline doc/03_reference/verification.rst:289]: LGTM

**Review by msfschaffner — APPROVED** [review]

**Review by moidx — APPROVED** [review]

**Comment by GregAC** [comment]
Updated to address @msfschaffner's comments. Also moved the checklists to their own page in the documentation.

**Comment by msfschaffner** [comment]
LGTM, thanks for the cleanup!

**Comment by GregAC** [comment]
A nightly regression run against latest master has hit our V2S goals

```
Legend: Metric* means cumulative e.g. Block* means Cumulative Block Coverage
name                                     Block* Covered       Branch* Covered      Statement* Covered   Expression* Covered  Toggle* Covered      Statement* Covered   Fsm* Covered         Assertion* Covered   CoverGroup* Covered  
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
uvm_pkg                                  n/a                  n/a                  n/a                  n/a                  n/a                  n/a                  n/a                  n/a                  100.00% (3/3)        
ibex_top                                 94.76% (12034/12700/1850) 90.76% (5511/6072/562) 94.72% (12437/13130/2294) 91.48% (4829/5279/4882) 98.71% (614/622/2402) 94.72% (12437/13130/2294) 100.00% (138/138)    98.08% (358/365/30)  41.39% (8159/19712)  
```

```
name                                     CoverGroup Average   CoverGroup Covered   
-----------------------------------------------------------------------------------
riscv_instr_pkg                          92.74%               92.87% (10523/11331) 
push_pull_agent_pkg                      100.00%              100.00% (3/3)        
core_ibex_fcov_if                        92.17%               49.76% (1247/2506)   
core_ibex_pmp_fcov_if.g_pmp_cgs_T        95.67%               90.70% (78/86)       
core_ibex_pmp_fcov_if.g_pmp_cgs_T.g_pmp_region_fcov 93.17%               76.43% (347/454)     
```
(Note we're using 'CoverGroup Average' for Ibex due to the large crosses meaning the 'CoverGroup Covered' is too biased towards those large crosses)

Regression pass rate:
```
93.04% PASS 1284 PASSED, 96 FAILED
```

Full regression log is attached: [regr.log](https://github.com/lowRISC/ibex/files/10044681/regr.log)

Merging this to declare V2 & V2S

**Comment by msfschaffner** [comment]
Awesome, congrats!

**Comment by moidx** [comment]
w0000t!

**Pull request merged**